src/common/toplvcmn.cpp

```diff
@@ -496,7 +496,13 @@ bool wxTopLevelWindowBase::Layout()
             int clientW, clientH;
             DoGetClientSize(&clientW, &clientH);
 
-            child->SetSize(0, 0, clientW, clientH);
+#ifndef __WXOSX_IPHONE__
```

Comment:
Can we please avoid this, either by using `GetClientAreaOrigin()` under all platforms if this is indeed the right thing to do or by adding another virtual function, e.g. `GetUsableAreaOrigin()` if it is not. But using such preprocessor checks is just not a good idea.